evm_arithmetization/benches/fibonacci_25m_gas.rs

```diff
@@ -0,0 +1,184 @@
+//! Benchmarks the CPU execution of a transaction calling a simple Fibonacci
+//! contract, iterating over and over until reaching the 25M gas limit.
+//!
+//! Total number of user instructions: 7_136_858.
+//! Total number of loops: 2_378_952.
+
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use env_logger::{try_init_from_env, Env, DEFAULT_FILTER_ENV};
+use ethereum_types::{Address, H256, U256};
+use evm_arithmetization::cpu::kernel::aggregator::KERNEL;
+use evm_arithmetization::cpu::kernel::opcodes::{get_opcode, get_push_opcode};
+use evm_arithmetization::generation::mpt::{AccountRlp, LegacyReceiptRlp};
+use evm_arithmetization::generation::{GenerationInputs, TrieInputs};
+use evm_arithmetization::proof::{BlockHashes, BlockMetadata, TrieRoots};
+use evm_arithmetization::prover::testing::simulate_execution;
+use evm_arithmetization::Node;
+use hex_literal::hex;
+use keccak_hash::keccak;
+use mpt_trie::nibbles::Nibbles;
+use mpt_trie::partial_trie::{HashedPartialTrie, PartialTrie};
+use plonky2::field::goldilocks_field::GoldilocksField;
+
+type F = GoldilocksField;
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let inputs = prepare_setup().unwrap();
+
+    // Dummy call to preinitialize the kernel.
+    let _ = KERNEL.hash();
+
+    let mut group = c.benchmark_group("fibonacci_25m_gas");
+    group.sample_size(10);
+    group.bench_function(BenchmarkId::from_parameter(8), |b| {
+        b.iter_batched(
+            || inputs.clone(),
+            |inp| simulate_execution::<F>(inp).unwrap(),
+            BatchSize::LargeInput,
+        )
+    });
+
+    // Last run to print the number of CPU cycles.
+    init_logger();
+    simulate_execution::<F>(inputs).unwrap();
```

Comment:
For some reason this line seems to be run whenever `cargo bench` is called, even with another benchmark than this one. Not sure how to solve this.

Reply:
Same as above, everything is fine with `--bench`.

Reply:
I can't reproduce this (which would be surprising tbh). How are you running the benchmarks?